docs/release/trg-1/trg-1-08.md

```diff
@@ -0,0 +1,28 @@
+---
+title: TRG 1.08 - Interface documentation (APIs)
+sidebar_position: 1
+---
+
+| Status | Created     | Post-History                                          |
+|--------|-------------|-------------------------------------------------------|
+| Active | 01-Aug-2024 |                                                       |
+| Draft  | 21-Mar-2024 | Transferred from Release Acceptance Criteria into TRG |
+
+## Why
+
+Interface documentation is crucial to interact with components. It fundamentally enables others to consume your APIs.
+
+## Description
+
+- The interface documentation **should** be stored in the folder `/docs/api` as `openAPI.yaml`
+  - It **must** be in the OpenAPI standard
+    - It **must** be at least version 2
+  - It **must** contain all interfaces
+  - It **must** be up-to-date for the latest released component version
+- Alternative the interface documentation **should** be referenced in the .tractusx meta file as reachable document (like GitHub release asset)
```

Comment:
this point should be merged with the first one, because I think the api-hub script always looks at the `.tractusx` meta file and never looks in the `docs/api` folder by itself

Reply:
Indeed in the current iteration it looks only into .tractux. Not sure if we really need the two options, seems the current apporach covers more flexibility, however I'm open for discussion. 
I'd also add reference to [metadata file docs](https://github.com/eclipse-tractusx/sig-release/blob/main/docs/metadata_file.md#openapi-specification) for more details on the OpenAPI specs urls configuration.

Reply:
The .tractusx file and the OpenAPI spec should have a 'convention over configuration' mode and should do something even if you don't configure / finetune it.

Reply:
the problem with picking up the file from the repository automatically is that will force contributors to keep there a copy of the specs aligned with the last version released, that could be potentially different from the one aligned with the `main` branch, but giving the possibility to configure the URL will permit to pick up the file from the release branch for example.
explicit configuration is always preferrable because it's more transparent, and will make the project contributors more aware of what's goin' on

Reply:
I disagree on 'explicit configuration...' The industry shifted to 'convention over configuration' in different areas to make things easier and working out of the box. But thats a philosophical question at this point. But we do the same for helm charts and other checks on github actions too.

You also shouldn't be able to merge anything API related without having a workflow verifying that your main branch has an up-to-date openapi spec. It would be wrong otherwise. But if you don't have it in the repo, we can't pick it up anyway. It doesn't hurt anyone who doesn't have it in place.

But it will allow everyone to browse through all APIs on the api-hub.

Reply:
nothing against the convention and having it as a fallback, but in fact it's very likely that if there's an openapi file checked in the repository, that would be aligned with the current main branch and not with the latest release (as it is stated in this document), so the typical case would be to have a link that points to the spec of the latest release (that could be either published somewhere or checked in the repository in a specific branch/tag)

Reply:
there should be a versioned spec anyway. main would/should probably only show up as next or main or something

Reply:
this would be up to the project, in case of specs generated out of the code there's no point in versioning it, because they are published somewhere else as artifacts (like we're doing with jars, docker images, charts).

Forcing projects to keep the file in the repository looks like an unnecessary constraint, in our case (tractusx-edc) will mean an additional burden to take care of.

Reply:
@ndr-brt i already updated the trg to have the 'alternativly...' this would exclude your case you just mentioned

Reply:
yep, my proposal was to move the "- Alternative..." line up near to the "- The interface documentation..." in a form like:
```
- The interface documentation **should** be either stored in the folder `/docs/api` as `openAPI.yaml` or referenced in the `.tractusx` meta file as reachable document (like GitHub release asset) through the `openApiSpecs` key
```

Reply:
@ndr-brt i wanted to keep the original sentence shorter but its better. I adjusted it.